app/controllers/assignments_controller.rb

```diff
@@ -37,20 +34,18 @@ def start
   # GET /assignments/new
   def new
     @assignment = Assignment.new
+    @assignment.group_id = params[:group_id]
     @assignment.deadline = Time.now+1.week
+    authorize @assignment, :admin_access?
   end
```

Comment:
I might have got this wrong, why not have `authorize @assignment, :admin_access?` as a before action hook? just like edit, update, destroy and reopen.

Reply:
Here I need an `@assignment` object before doing an authorize, in the new method `@assignment` is not already set like in edit, update, destroy and reopen because it doesn't exist in the database yet. I have to still create it. So I have to create a dummy object like so `@assignment = Assignment.new` and assign the group_id to do the respective authorize.

Reply:
Okay